content/usgs-packages/Application_Part2.Rmd

```diff
@@ -0,0 +1,138 @@
+---
+title: "Application - Part 2, download data"
+date: "9999-05-01"
+author: "Lindsay R. Carr"
+slug: "app-part2"
+image: "img/main/intro-icons-300px/r-logo.png"
+output: USGSmarkdowntemplates::hugoTraining
+parent: Introduction to USGS R Packages
+weight: 2
+draft: true
+---
+
+```{r setup, include=FALSE, warning=FALSE, message=FALSE}
+library(knitr)
+
+knit_hooks$set(plot=function(x, options) {
+  sprintf("<img src='../%s%s-%d.%s'/ title='%s'/>", 
+          options$fig.path, options$label, options$fig.cur, options$fig.ext, options$fig.cap)
+
+})
+
+opts_chunk$set(
+  echo=TRUE,
+  fig.path="static/app-part2/",
+  fig.width = 6,
+  fig.height = 6,
+  fig.cap = "TODO"
+)
+
+knit_hooks$set(addToggle = function(before, options, envir) {
+    if(before) {
+      sprintf('<button class="ToggleButton" onclick="toggle_visibility(\'%1$s\')">Show Answer</button>
+              <div id="%1$s" style="display:none">', opts_current$get('label'))
+    } else {
+      '</div>'
+    }
+})
+
+set.seed(1)
+```
+
+```{r sbtools-auth, echo=FALSE}
+# run vizlab::storeSBcreds() once before this can work
+credList <- readRDS(file.path(path.expand('~'), ".vizlab/sbCreds"))
+un <- rawToChar(credList$username)
+pw <- rawToChar(credList$password)
+sbtools::authenticate_sb(un, pw)
+```
+
+In this section, we are going to use `dataRetrieval` and `geoknife` to get nitrogen, phosphorus, and precipitation data for the sites determined in the [previous section](/usgs-packages/app-part1).
+
+We are walking through the workflow in very distinct chunks, but this will be put together as a single script at the end. If you need a reminder, the code that we used to get the site and 8-digit HUC numbers is available below.
+
+```{r get-sb-sites, addToggle="ON", message=FALSE, warning=FALSE}
+library(sbtools)
+library(dataRetrieval)
+
+# identify site id and query for files
+sb_site_id <- "59848b35e4b0e2f5d46717d1"
+avail_files <- item_list_files(sb_site_id)
+
+# use appropriate reader to get file (tab delimited) into R & get site numbers
+sb_sites_df <- read.table(avail_files$url[1], sep="\t", header=TRUE,
+                          colClasses = "character", stringsAsFactors = FALSE)
+sites <- sb_sites_df$site_number
+
+# get HUC 8 codes for precip data
+sb_sites_info <- readNWISsite(sites)
+huc8s <- sb_sites_info$huc_cd
+```
+
+Before downloading the data, make sure you identify the time period of interest. For this example, we are going to use water year 2016.
+
+```{r setup-time-period}
+startDate <- "2015-10-01"
+endDate <- "2016-09-30"
+```
+
+## Get nutrient data
+
+Now, use `dataRetrieval` functions to pull down data for nitrogen and phosphorus. You can choose your own parameter codes to define these nutrients using `parameterCdFile` or use the ones below.
+
+```{r nutrient-pcodes}
+pcodes_nitrogen <- c("00613", "00618", "00631")
```

Comment:
i like that you didn't hide this chunk behind a Show Answer button. will save students a headache